lib/unexpectedMitm.js

```diff
@@ -1047,22 +1052,30 @@ module.exports = {
                                  * reject the assertion. This is only safe with
                                  * Unexpected 10.15.x and above.
                                  */
+
+                                exchangeParts.push(e);
                                 reject(e);
                             }
                         });
 
                         function assertMockResponse(mockResponse, mockResponseError) {
                             var mockRequest = new messy.HttpRequest(requestProperties);
+                            var httpPart = {
+                                exchange: undefined,
+                                spec: undefined
+                            };
+
                             var httpExchange = new messy.HttpExchange({request: mockRequest});
```

Comment:
I think I'd set this directly as the `exchange` property above, but it's certainly a matter of taste.